lib/charms/mongodb/v0/upgrade.py

```diff
@@ -65,30 +82,91 @@ def pre_upgrade_check(self) -> None:
             raise ClusterNotReadyError(message=default_message, cause="Cluster is not healthy")
 
         if not self.is_cluster_able_to_read_write():
-            raise ClusterNotReadyError(message=default_message, cause="Cluster cannot read/write")
+            raise ClusterNotReadyError(
+                message=default_message, cause="Cluster cannot read/write - please check logs"
```

Comment:
**suggestion:** I believe that `ClusterNotReadyError` also takes `resolution` as an arg. I think here - and in some of the others - you can help the user along with a resolution recommendation, e.g 'Run the check on the config leader' or whatever.

Reply:
great suggestion! I will add wherever I can provide useful resolutions :)